storage/static.go

```diff
@@ -25,6 +26,9 @@ type staticClientsStorage struct {
 func WithStaticClients(s Storage, staticClients []Client) Storage {
 	clientsByID := make(map[string]Client, len(staticClients))
 	for _, client := range staticClients {
+		if re, err := regexp.Compile("^" + strings.Join(client.RedirectURIs, "|") + "$"); err == nil {
```

Comment:
The regexp built here is missing a caption group  over the joined `RedirectURIs` causing it to match two concatenated URIs.

Given two RedirectURIs `http://foo.com` and `http://bar.com` the compiled regex would look like 

```
^http://foo.com|http.bar.com$
```

and it would match 

```
http://foo.comhttp://bar.com
```

which doesn't look right to me.

Instead the regexp should look like `^(http://foo.com|http://bar.com)$`